src/proposals/mips/mip-b31/mip-b31.sol

```diff
@@ -0,0 +1,107 @@
+import {MultiRewardDistributorCommon} from "@protocol/rewards/MultiRewardDistributorCommon.sol";
+
+/// DO_PRE_BUILD_MOCK=true DO_VALIDATE=true DO_PRINT=true DO_BUILD=true DO_RUN=true forge script
+/// src/proposals/mips/mip-b31/mip-b31.sol:mipb31
+contract mipb31 is HybridProposal, Configs {
+    string public constant override name = "MIP-B31";
+
+    uint256 public constant SUPPLY_SIDE_REWARDS = 4538;
+
+    uint256 public constant BORROW_SIDE_REWARDS = 3025;
+
+    uint256 public constant END_TIME = 24994202400;
+
+    constructor() {
+        bytes memory proposalDescription = abi.encodePacked(
+            vm.readFile("./src/proposals/mips/mip-b31/MIP-B31.md")
+        );
+
+        _setProposalDescription(proposalDescription);
+    }
+
+    function primaryForkId() public pure override returns (uint256) {
+        return BASE_FORK_ID;
+    }
+
+    function deploy(Addresses addresses, address) public override {}
+
+    function afterDeploy(Addresses addresses, address) public override {}
+
+    function preBuildMock(Addresses addresses) public override {}
+
+    function build(Addresses addresses) public override {
+        address market = addresses.getAddress("MOONWELL_EURC");
+        address mrd = addresses.getAddress("MRD_PROXY");
+        address emissionToken = addresses.getAddress("EURC");
+
+        _pushAction(
+            mrd,
+            abi.encodeWithSignature(
+                "_addEmissionConfig(address,address,address,uint256,uint256,uint256)",
+                MToken(market),
+                addresses.getAddress("TEMPORAL_GOVERNOR"),
+                emissionToken,
+                SUPPLY_SIDE_REWARDS,
+                BORROW_SIDE_REWARDS,
+                END_TIME
+            ),
+            "Add emission config for Moonwell EURC on Base"
+        );
+    }
+
+    function teardown(Addresses addresses, address) public pure override {}
+
+    /// @notice assert that the new interest rate model is set correctly
+    /// and that the interest rate model parameters are set correctly
+    function validate(Addresses addresses, address) public view override {
+        address market = addresses.getAddress("MOONWELL_EURC");
+        address emissionToken = addresses.getAddress("EURC");
+        address mrd = addresses.getAddress("MRD_PROXY");
+
+        MultiRewardDistributorCommon.MarketConfig
+            memory config = IMultiRewardDistributor(mrd).getConfigForMarket(
+                MToken(market),
+                emissionToken
+            );
+
+        assertEq(
+            config.supplyEmissionsPerSec,
+            SUPPLY_SIDE_REWARDS,
+            "Supply rewards not set correctly"
+        );
+        assertEq(
+            config.borrowEmissionsPerSec,
+            BORROW_SIDE_REWARDS,
+            "Borrow rewards not set correctly"
+        );
+        assertEq(config.endTime, END_TIME, "End time not set correctly");
+
+        uint256 startTimestamp = 1726695000;
```

Comment:
we should set the start time to `1727127000` which is Mon Sep 23 2024 14:30:00 GMT-0700 (Pacific Daylight Time), that way we'll be precise.

I forgot we can do this, we just need to propose sometime before 2:30pm Pacific today and it will begin exactly at the right time. Is there a risk if the proposal doesn't execute quickly enough? Start time isn't required to be in the future is it?